.github/workflows/ci.yaml

```diff
@@ -1,6 +1,15 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
+    branches:
+      - "**"
+
+  pull_request:
+    branches:
+      - "**:**"
```

Comment:
I fear this is always getting more verbose.

Code reuse is hard in workflows, isn't it?

Reply:
Yes! Hopefully the list of events won't change much.

But it was much simpler with CircleCI (thanks to `&`). Or maybe we must change how we look at it, and must write actual actions, that could be shared across repositories?

Reply:
Maybe something in the spirit of manastech/ci-docker-builder but that also checks whether to run the tests or not.

Like, having the CI-provider-specific workflow be "get this script, always run these tests, then build me a docker image", and have the script (shared between most repos) decide when to run the workflow (depending on branch names, tag names, etc).

🤷

Reply:
Yes. Exactly!